packages/web-api/src/controllers/scan-request-controller.ts

```diff
@@ -73,27 +74,38 @@ export class ScanRequestController extends ApiController {
 
         // tslint:disable-next-line: no-null-keyword
         this.logger.trackEvent('BatchScanRequestSubmitted', null, measurements);
+        let priority: string;
+        for (priority of Object.keys(processedData.requestCountByPriority)) {
+            this.logger.trackMetric('QueuedScanRequests', processedData.requestCountByPriority[priority], { priority: priority });
+        }
     }
 
     private getProcessedRequestData(batchId: string, scanRunRequests: ScanRunRequest[]): ProcessedBatchRequestData {
         const scanRequestsToBeStoredInDb: ScanRunBatchRequest[] = [];
         const scanResponses: ScanRunResponse[] = [];
+        const requestCountByPriority: { [priority: string]: number } = {};
 
         scanRunRequests.forEach(scanRunRequest => {
             const runRequestValidationResult = this.validateRunRequest(scanRunRequest);
             if (runRequestValidationResult.valid) {
                 // preserve GUID origin for a single batch scope
                 const scanId = this.guidGenerator.createGuidFromBaseGuid(batchId);
+                const priority = isNil(scanRunRequest.priority) ? 0 : scanRunRequest.priority;
                 scanRequestsToBeStoredInDb.push({
                     scanId: scanId,
-                    priority: isNil(scanRunRequest.priority) ? 0 : scanRunRequest.priority,
+                    priority: priority,
                     url: scanRunRequest.url,
                 });
 
                 scanResponses.push({
                     scanId: scanId,
                     url: scanRunRequest.url,
                 });
+
+                if (requestCountByPriority[priority.toString()] === undefined) {
+                    requestCountByPriority[priority.toString()] = 0;
```

Comment:
Should we count first one ie set to 1

Reply:
I'm wondering if we can use [splitting ](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-charts#apply-filters-to-charts) feature so that we can split the data by priority in app insights instead of doing it here.